packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -749,12 +748,97 @@ const Set<String> _packageDependencies = <String>{
   'yaml',
 };
 
+// A valid Dart identifier.
+// https://dart.dev/guides/language/language-tour#important-concepts
+final RegExp _identifierRegExp = RegExp('[a-zA-Z_][a-zA-Z0-9_]*');
+
+// non-contextual dart keywords.
+//' https://dart.dev/guides/language/language-tour#keywords
+const Set<String> _keywords = <String>{
```

Comment:
This is a _little_ scary - who will update this as new keywords get added?

Reply:
I'm not actually sure if pub can add more restrictions here. Grammatically, there are no issues since package-scheme imports are just URIs, and `package:abstract/abstract.dart` is in no way ambiguous. 

But the analyzer function we were using before wasn't the pub source of truth either, and in the worst case `pub get` fails with an error message that they must change the package name (only in a single place).

Reply:
Also, the pub restrictions only matter if you publish to pub - which for applications is somewhat pointless